components/tikv_alloc/src/profiler_guard.rs

```diff
@@ -0,0 +1,36 @@
+use super::error::ProfResult;
+use super::{activate_prof, deactivate_prof};
+
+use futures::{Future, Poll};
+use futures_locks::{Mutex, MutexFut, MutexGuard};
+
+lazy_static! {
+    static ref PROFILER_MUTEX: Mutex<u32> = Mutex::new(0);
+}
+
+pub struct ProfilerLock(MutexFut<u32>);
```

Comment:
I prefer ProfGuard to keep the same with ProfResult

Reply:
Sorry, I don't understand what you mean.

Reply:
He meant to rename it from `Profiler*` to `Prof*`.